hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/metrics/ContainerHealthMetrics.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.metrics;
+
+import org.apache.hadoop.hdds.annotation.InterfaceAudience;
+import org.apache.hadoop.metrics2.MetricsSystem;
+import org.apache.hadoop.metrics2.annotation.Metric;
+import org.apache.hadoop.metrics2.annotation.Metrics;
+import org.apache.hadoop.metrics2.lib.DefaultMetricsSystem;
+import org.apache.hadoop.metrics2.lib.MutableGaugeLong;
+import org.apache.hadoop.ozone.OzoneConsts;
+
+/**
+ * Class for tracking metrics related to container health task in Recon.
+ */
+@InterfaceAudience.Private
+@Metrics(about = "Recon ContainerHealthTask Metrics", context = OzoneConsts.OZONE)
+public final class ContainerHealthMetrics {
+
+  private static final String SOURCE_NAME =
+      ContainerHealthMetrics.class.getSimpleName();
+
+  private ContainerHealthMetrics() {
+  }
+
+  public static ContainerHealthMetrics create() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    return ms.register(SOURCE_NAME,
+        "Recon Container Health Task Metrics",
+        new ContainerHealthMetrics());
+  }
+
+  public void unRegister() {
+    MetricsSystem ms = DefaultMetricsSystem.instance();
+    ms.unregisterSource(SOURCE_NAME);
+  }
+
+  @Metric(about = "Number of missing containers detected in Recon.")
+  private MutableGaugeLong missingContainerCount;
+
+  @Metric(about = "Number of under replicated containers detected in Recon.")
+  private MutableGaugeLong underReplicatedContainerCount;
+
+  @Metric(about = "Number of corrupted containers detected in Recon.")
+  private MutableGaugeLong corruptedContainerCount;
+
+  public void setMissingContainerCount(long missingContainerCount) {
+    this.missingContainerCount.set(missingContainerCount);
+  }
+
+  public void setUnderReplicatedContainerCount(long underReplicatedContainerCount) {
+    this.underReplicatedContainerCount.set(underReplicatedContainerCount);
+  }
+
+  public void setCorruptedContainerCount(long corruptedContainerCount) {
+    this.corruptedContainerCount.set(corruptedContainerCount);
```

Comment:
I couldn't find a reference to this. Do we plan to use this metric in the future?

Reply:
Yes, I am planning to use this and update once this [PR](https://github.com/apache/ozone/pull/7942) gets merged.

Reply:
We should use the term mismatch, not corruption, to handle mismatched replicas. Containers may have different checksums without having any corrupted data.

Reply:
Thanks @errose28 for the review. I have updated the term.